sync15-adapter/src/state.rs

```diff
@@ -2,15 +2,42 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
+use std::mem;
 
 use bso_record::BsoRecord;
+use client::SetupStorageClient;
 use collection_keys::CollectionKeys;
 use error::{self, ErrorKind};
 use key_bundle::KeyBundle;
-use record_types::MetaGlobalRecord;
+use record_types::{MetaGlobalEngine, MetaGlobalRecord};
 use request::InfoConfiguration;
-use util::{ServerTimestamp, SERVER_EPOCH};
+use util::{random_guid, ServerTimestamp, SERVER_EPOCH};
+
+use self::SetupState::*;
+
+const STORAGE_VERSION: usize = 5;
+
+lazy_static! {
+    /// Maps names to storage versions for engines to include in a fresh
+    /// `meta/global` record.
+    static ref DEFAULT_ENGINES: Vec<(&'static str, usize)> = vec![
```

Comment:
Yeah, I'm inclined to agree with Nick here. This patch appears to be making this the basis for a "sync client" in the same flavour that Firefox has - but I'm not sure that's where we want to head. OTOH though, I'm also not clear that it would be wrong to do that - this code is going to be "per app" - ie, there's no way 2 android apps are going to share the same DB and state, and I can't see that a single app would ever want 2 discrete password sync engines.

I think that if we can arrange to pass the state around we can sidestep these questions for now, although I haven't thought too much about what that would look like. I wonder if we should have a meeting to discuss this?